src/libstd/macros.rs

```diff
@@ -210,21 +217,15 @@ macro_rules! debug_assert_eq {
 /// }
 /// ```
 #[macro_export]
+#[stable]
 macro_rules! unreachable {
-    () => ({
-        panic!("internal error: entered unreachable code")
-    });
-    ($msg:expr) => ({
-        unreachable!("{}", $msg)
-    });
-    ($fmt:expr, $($arg:tt)*) => ({
-        panic!(concat!("internal error: entered unreachable code: ", $fmt), $($arg)*)
-    });
+    () => (panic!("internal error: entered unreachable code"))
```

Comment:
I'd personally like to remove both of them as they're just longer versions of `panic!` in my mind, but failing that I'd rather stick to a more conservative approach where we don't just duplicate-panic-in-each-macro just yet.